REGRESSION(252485@main): Restore previous usage of pthred_main_np()

aperezdc opened the pull request:
#### 0307ab9b687a20115e7c2200f58ecaca0d268afd
<pre>
REGRESSION(252485@main): Restore previous usage of pthred_main_np()
<a href="https://bugs.webkit.org/show_bug.cgi?id=242690">https://bugs.webkit.org/show_bug.cgi?id=242690</a>

Reviewed by Darin Adler.

* Source/WTF/wtf/generic/MainThreadGeneric.cpp:
(WTF::isMainThread): Partially revert 252485@main to restore the
previous usage of pthread_main_np(), which should have not been
changed.

Canonical link: <a href="https://commits.webkit.org/252601@main">https://commits.webkit.org/252601@main</a>
</pre>

jameshilliard commented on Source/WTF/wtf/generic/MainThreadGeneric.cpp:
```diff
@@ -61,9 +61,7 @@ void initializeMainThreadPlatform()
 {
 #if HAVE(PTHREAD_MAIN_NP)
-    int mainThreadNp = pthread_main_np();
-    ASSERT(mainThreadNp != -1);
```
Think we should leave the assert in?

aperezdc replied:
I had the same doubt, but to be on the safe side I ended up leaving this as it was before. Hopefully @darinadler would comment on this later.

(My interpretation is that maybe the original code wanted the `-1` result, which means “threads are being initialized”, to be the same as “I am the main thread, because there are no other threads yet.)

darinadler replied:
I think it’s fine to leave the assertion in.

darinadler replied:
But also OK to remove it as we have done here.

aperezdc replied:
I think that I prefer to leave it as-is without the assertion, to avoid breaking accidentally the BSD systems which may return `-1` and that we do not actively test ourselves. Thanks!

aperezdc committed:
REGRESSION(252485@main): Restore previous usage of pthred_main_np()
https://bugs.webkit.org/show_bug.cgi?id=242690

Reviewed by Darin Adler.

* Source/WTF/wtf/generic/MainThreadGeneric.cpp:
(WTF::isMainThread): Partially revert 252485@main to restore the
previous usage of pthread_main_np(), which should have not been
changed.

Canonical link: https://commits.webkit.org/252601@main
